nixos/modules/flake/source-info.nix

```
@@ -0,0 +1,31 @@
+{ config, lib, ... }:
+
+with lib;
+
+let sourceInfo = config.system.nixos.configuration.sourceInfo;
+
+in
+{
+  options = {
+
+    system.nixos.configuration.sourceInfo = mkOption {
+      type = types.attrsOf types.anything;
+      default = {};
+      example = "self.sourceInfo";
+      description = ''
```

You could do this and then write markdown if you like

```suggestion
      description = lib.mdDoc ''
```

---

Updated. The example also gets improved.